okapi-core/src/main/java/org/folio/okapi/managers/TenantManager.java

```diff
@@ -344,8 +344,33 @@ private void waitTenantInit(Tenant tenant, ModuleInstance getInstance,
    * @return fut future
    */
   private Future<Void> invokeTenantInterface(Tenant tenant, TenantInstallOptions options,
-                                             ModuleDescriptor mdFrom, ModuleDescriptor mdTo,
-                                             ProxyContext pc) {
+      ModuleDescriptor mdFrom, ModuleDescriptor mdTo,
+      ProxyContext pc) {
+
+    Future<Void> future;
+    if (mdTo != null && options.getPurge()) {
+      // enable with purge is turned into purge on its own, followed by regular enable
+      mdFrom = null;
+      future = invokeTenantInterface1(tenant, options, mdTo, null, pc)
```

Comment:
Is it intended that not the old module (mdFrom) but the new module (mdTo) executes the purge?

Reply:
Yes

Reply:
The from is irrelevant and discarded because the idea is to reset the module to that version initialized (mdTo .. id property in install). Suppose the module was already running in some other version (according to Okapi), that information is explicitly reset.